flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java

```diff
@@ -459,7 +459,10 @@ public static String[] mergeListsToArray(List<String> base, List<String> append)
             ConfigOptions.key("parallelism.default")
                     .intType()
                     .defaultValue(1)
-                    .withDescription("Default parallelism for jobs.");
+                    .withDescription("Default parallelism for jobs. There is an exception here. When creating a"
```

Comment:
I am not sure what the new description is saying:
- Is the description ever a default of 1? If the default is not always 1 - I suggest we consider removing the default though this would have migration implications we would need to think through. 
- the phrase "There is an exception here." I would not use the word exception; as this is an overloaded word implying there might be an error of some sort.
- what does "the return value of Runtime. getRuntime. availableProcessors()" mean?

Reply:
@beliefer when you say special cases, do you mean that there are 3 scenarios , default = 1 , case 1 and case 2. If default=1 never happens - then we could remove it.